src/stories/lib/smart-auth-plugin.js

```diff
@@ -0,0 +1,61 @@
+export const injectAuthForSmartComponentsPlugin = {
+  name: 'inject-auth-for-smart-components',
+  async transform (code, id) {
+    const SMART_COMPONENT_STORY_REGEX = /\/src\/components\/.*\/cc-.*\.smart.*\.md$/;
+
+    if (SMART_COMPONENT_STORY_REGEX.test(id)) {
+      const {
+        API_HOST,
+        API_OAUTH_TOKEN,
+        API_OAUTH_TOKEN_SECRET,
+        OAUTH_CONSUMER_KEY,
+        OAUTH_CONSUMER_SECRET,
+      } = process.env;
+
+      // language=JavaScript
+      code = `
+        import { updateRootContext } from '../../lib/smart-manager.js';
+        import '${id.replace('.md', '.js')}';
+
+        ${code}
+
+        updateRootContext({
+          apiConfig: {
+            API_HOST: '${API_HOST}',
+            API_OAUTH_TOKEN: '${API_OAUTH_TOKEN}',
+            API_OAUTH_TOKEN_SECRET: '${API_OAUTH_TOKEN_SECRET}',
+            OAUTH_CONSUMER_KEY: '${OAUTH_CONSUMER_KEY}',
+            OAUTH_CONSUMER_SECRET: '${OAUTH_CONSUMER_SECRET}',
+          },
+        });
+      `;
+      return code;
+    }
+
+    if (id.includes('/demo-smart/index.js')) {
+      const {
+        WARP_10_HOST,
+        API_HOST,
+        API_OAUTH_TOKEN,
+        API_OAUTH_TOKEN_SECRET,
+        OAUTH_CONSUMER_KEY,
+        OAUTH_CONSUMER_SECRET,
+      } = process.env;
+
+      code += `
+        updateRootContext({
+          apiConfig: {
+            WARP_10_HOST: '${WARP_10_HOST}',
+            API_HOST: '${API_HOST}',
+            API_OAUTH_TOKEN: '${API_OAUTH_TOKEN}',
+            API_OAUTH_TOKEN_SECRET: '${API_OAUTH_TOKEN_SECRET}',
+            OAUTH_CONSUMER_KEY: '${OAUTH_CONSUMER_KEY}',
+            OAUTH_CONSUMER_SECRET: '${OAUTH_CONSUMER_SECRET}',
+          },
+        });
+      `;
+
+      return code;
+    }
```

Comment:
suggestion: can we factorize this a bit and simplify the code? The only difference seems to be Warp10 and (for now) both could/should use it.